enhancing/enhancer.js

```diff
@@ -2,21 +2,36 @@ module.exports = {
   succeed,
   fail,
   repair,
-  get,
+  get
 };
 
 function succeed(item) {
-  return { ...item };
+  return {
+    ...item,
+    enhancement: item.enhancement < 20 ? item.enhancement + 1 : item.enhancement
+  };
 }
 
 function fail(item) {
-  return { ...item };
+  return {
+    ...item,
+    durability:
+      item.enhancement < 15 ? item.durability - 5 : item.durability - 10,
+    enhancement: item.enhancement > 16 ? item.enhancement - 1 : item.enhancement
+  };
 }
 
 function repair(item) {
-  return { ...item };
+  return { ...item, durability: 100 };
 }
 
 function get(item) {
-  return { ...item };
+  if (item.enhancement === 0) {
+    return item;
+  } else {
+    return {
+      ...item,
+      name: `[+${item.enhancement}] ${item.name}`
```

Comment:
another extra edge case! (just because these are fun to think of) what if an item has already been modified this way and is being passed through again? for example, you'd get something like `[+7] [+7] iron dagger` returned as it is now